fcrepo-kernel-api/src/test/java/org/fcrepo/kernel/api/utils/RelaxedPropertiesHelperTest.java

```diff
@@ -0,0 +1,295 @@
+            ),
+            Triple.create(
+                subject,
+                CONTAINS.asNode(),
+                NodeFactory.createLiteral("some-value")
+            )
+        );
+        for (final var triple : triples) {
+            assertThrows(ServerManagedPropertyException.class,
+                    () -> RelaxedPropertiesHelper.checkTripleForDisallowed(triple));
+        }
+    }
+
+    /**
+     * Test certain predicates that are relaxable. These are
+     * - <a href="http://fedora.info/definitions/v4/repository#createdBy">createdBy</a>
+     * - <a href="http://fedora.info/definitions/v4/repository#createdDate">createdDate</a>
+     * - <a href="http://fedora.info/definitions/v4/repository#lastModifiedBy">lastModifiedBy</a>
+     * - <a href="http://fedora.info/definitions/v4/repository#lastModifiedDate">lastModifiedDate</a>
+     *
+     */
+    @Test
+    public void testRelaxablePredicate() {
+        final List<Triple> triples = List.of(
+            Triple.create(
+                subject,
+                CREATED_BY.asNode(),
+                NodeFactory.createLiteral("some-value")
+            ),
+            Triple.create(
+                subject,
+                CREATED_DATE.asNode(),
+                NodeFactory.createLiteral("some-value")
+            ),
+            Triple.create(
+                subject,
+                LAST_MODIFIED_BY.asNode(),
+                NodeFactory.createLiteral("some-value")
+            ),
+            Triple.create(
+                subject,
+                LAST_MODIFIED_DATE.asNode(),
+                NodeFactory.createLiteral("some-value")
+            )
+        );
+        for (final var triple : triples) {
+            assertThrows(RelaxableServerManagedPropertyException.class,
+                    () -> RelaxedPropertiesHelper.checkTripleForDisallowed(triple));
+        }
+    }
+
+    /**
+     * Test that mime-types sent in RDF with a ebucore:hasMimeType predicate must be valid or a variable.
+     * @see <a href="https://tools.ietf.org/html/rfc6838">RFC 6838</a>
+     */
+    @Test
+    public void testValidMimeType() {
+        final List<Triple> valid_triples = List.of(
+                Triple.create(
+                        subject,
+                        HAS_MIME_TYPE.asNode(),
+                        NodeFactory.createLiteral("text/plain")
+                ),
+                Triple.create(
+                        subject,
+                        HAS_MIME_TYPE.asNode(),
+                        NodeFactory.createLiteral("application/json")
+                ),
+                Triple.create(
+                        subject,
+                        HAS_MIME_TYPE.asNode(),
+                        NodeFactory.createLiteral("application/n-triples")
+                ),
+                Triple.create(
+                        subject,
+                        HAS_MIME_TYPE.asNode(),
+                        NodeFactory.createVariable("some-variable")
```

Comment:
I'm honestly not totally sure what this means. Does it imply that if someone used a sparql variable to supply "not-a-mimetype" as the mimetype, that it would be accepted by fedora?

Reply:
This is used to parse both incoming RDF as well as Sparql. So you can have a triple with a HAS_MIME_TYPE predicate and a variable. We can restrict it, I'm not sure if there is a valid use case I just didn't want to go overboard.

Reply:
well either way we don't need to restrict it in this PR